probe_src/python/probe_py/manual/analysis.py

```diff
@@ -199,6 +234,127 @@ def add_edges(edges:list[tuple[Node, Node]], label:EdgeLabels) -> None:
     add_edges(fork_join_edges, EdgeLabels.FORK_JOIN)
     return process_graph
 
+def traverse_hb_for_dfgraph(process_tree_prov_log: ProvLog, starting_node: Node, traversed: set[int] , dataflow_graph:nx.DiGraph, file_version_map: Dict[InodeOnDevice, int], shared_files: set[InodeOnDevice], cmd_map: Dict[int, list[str]]) -> None:
+    starting_pid = starting_node[0]
+    
+    starting_op = prov_log_get_node(process_tree_prov_log, starting_node[0], starting_node[1], starting_node[2], starting_node[3])
+    process_graph = provlog_to_digraph(process_tree_prov_log)
+    
+    edges = list_edges_from_start_node(process_graph, starting_node)
+    name_map = collections.defaultdict[InodeOnDevice, list[pathlib.Path]](list)
+
+    target_nodes = collections.defaultdict[int, list[Node]](list)
+    console = rich.console.Console(file=sys.stderr)
+    
+    for edge in edges:  
+        pid, exec_epoch_no, tid, op_index = edge[0]
+        
+        # check if the process is already visited when waitOp occurred
+        if pid in traversed or tid in traversed:
+            continue
+        op = prov_log_get_node(process_tree_prov_log, pid, exec_epoch_no, tid, op_index).data
+        next_op = prov_log_get_node(process_tree_prov_log, edge[1][0], edge[1][1], edge[1][2], edge[1][3]).data
+        if isinstance(op, OpenOp):
+            access_mode = op.flags & os.O_ACCMODE
+            processNode = ProcessNode(pid=pid, cmd=''.join(cmd_map[pid]))
```

Comment:
Let's store `cmd_map[pid]` (list of strings) instead of as a joined string.

Reply:
I tried doing this before, but since a list is not hashable, an error occurred when calling `dataflow_graph.add_node(processNode)`